docs: improve regexManager documentation (#19278)
renovatebot/renovate: lib/modules/manager/regex/readme.md

The syntax in the example is arbitrary and you can set your own syntax.
If you do, update your `matchStrings` regex!

### Using regexManager to update the dependency name in addition to version

#### Updating `gitlab-ci include` dep names

> If the location of files used in gitlab-ci `includes:` has changed you can use the regex manager to update the `depName` as well as the version.
> 
> When doing so you may find you need to specify a second `matchString` for the new name, as when the regex manager validates the change it will not always detect that the new value is valid.

Reviewer: ```suggestion
You can use the regex manager to update the `depName` and the version.
This can be handy when the location of files referenced in gitlab-ci `includes:` fields has changed.

You may need to set a second `matchString` for the new name to ensure the regex manager can detect the new value.
For example:
```

```json
{
  "regexManagers": [
    {
      "fileMatch": [".*y[a]?ml$"],
      "matchStringsStrategy": "combination",
      "matchStrings": [
        "['\"]?(?<depName>/pipeline-fragments\\/fragment-version-check)['\"]?\\s*ref:\\s['\"]?(?<currentValue>[\\d-]*)['\"]?",
        "['\"]?(?<depName>pipeline-solutions\\/gitlab\\/fragments\\/fragment-version-check)['\"]?\\s*ref:\\s['\"]?(?<currentValue>[\\d-]*)['\"]?"
      ],
      "depNameTemplate": "pipeline-solutions/gitlab/fragments/fragment-version-check",
      "autoReplaceStringTemplate": "'{{{depName}}}'\n    ref: {{{newValue}}}",
      "datasourceTemplate": "gitlab-tags",
      "versioningTemplate": "gitlab-tags"
    }
  ]
}
```

> Will migrate:

Reviewer: ```suggestion
The config above will migrate:
```

```yaml
- project: 'pipeline-fragments/docker-lint'
  ref: 2-4-0
  file: 'ci-include-docker-lint-base.yml'
```

> to

Reviewer: ```suggestion
To this:
```
